src/pretix/base/services/mail.py

```diff
@@ -466,9 +466,17 @@ def mail_send_task(self, *args, to: List[str], subject: str, body: str, html: st
             for inv in invoices:
                 if inv.file:
                     try:
+                        # We try to give the invoice a more human-readable name, e.g. "Invoice_ABC-123.pdf" instead of
+                        # just "ABC-123.pdf", but we only do so if our currently selected language allows to do this
+                        # as ASCII text. For example, we would not want a "فاتورة_" prefix for our filename since this
+                        # has shown to cause deliverability problems of the email and deliverability wins.
+                        filename = pgettext('invoice', 'Invoice {num}').format(num=inv.number).replace(' ', '_') + '.pdf'
+                        if not re.match("^[a-zA-Z0-9-_%./,&:# ]+$", filename):
+                            filename = inv.number.replace(' ', '_') + '.pdf'
+                        filename = re.sub("[^a-zA-Z0-9-_.]", "_", filename)
```

Comment:
Should multiple occurrences in a row only be replaced by a single underscore? Could be done to the spaces-replacement as well, but there it is probably overkill.

```suggestion
                        filename = re.sub("[^a-zA-Z0-9-_.]+", "_", filename)
```